### PR TITLE
Added pre-commit for linting purposes. Fixes #7476

### LINKED ISSUE #7476: Install pre-commit to avoid un-linted/unformatted code from being committed
## Is this a BUG / ISSUE report or a QUESTION?

Enhancement

## Describe the problem you're observing.

As a contributor, I've passed in several commits that haven't been linted/formatted after forgetting to run ```black``` and ```flake8```. This could be avoided by running pre-commit hooks to analyze the code before the commit is made thus circumventing unformatted code from being committed.

### PR DESCRIPTION
**Pull request by sashadev-sky**
Includes changes from #7477

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+repos:
+-   repo: https://github.com/psf/black
+    rev: 23.1.0
+    hooks:
+    -   id: black
```
Comment (ThomasWaldmann) [inline .pre-commit-config.yaml:5]: we won't backport the `black` releated part of this and won't globally change src formatting in 1.2 branch.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+-   repo: https://github.com/pycqa/flake8
+    rev: 6.0.0
+    hooks:
+    -   id: flake8
+        files: '(src|scripts|conftest.py)'
```
Comment (ThomasWaldmann) [inline .pre-commit-config.yaml:10]: good, we'll take this for 1.2.

File: docs/development.rst [hunk 3]
```diff
@@ -164,6 +164,13 @@ virtual env and run::
   pip install -r requirements.d/development.txt
 
 
+This project utilizes pre-commit to format and lint code before it is committed.
```
Comment (ThomasWaldmann) [inline docs/development.rst:167]: remove "format and ".

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]
thanks for backporting this.

sorry, maybe i missed to mention it, but the `black` part should not be part of the backport.

**Comment by sashadev-sky** [comment]
@ThomasWaldmann updated!

**Review by ThomasWaldmann — APPROVED** [review]
LGTM!

**Pull request merged**